### PR TITLE
Implemented Login UI

### PR DESCRIPTION
**Pull request by AndreeaBogdan**

### CODE REVIEW

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    @IBOutlet weak var loginButton: UIButton!
+    @IBAction func loginButton(_ sender: UIButton) {
+ 
+        // validation for empty email textfield
+        guard let email = emailTextField.text, emailTextField.text?.count != 0
+        else{
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:26]: this should be in previous line: "guard condition else { "

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    @IBOutlet weak var validationLabel: UILabel!
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    @IBOutlet weak var loginButton: UIButton!
+    @IBAction func loginButton(_ sender: UIButton) {
+ 
+        // validation for empty email textfield
+        guard let email = emailTextField.text, emailTextField.text?.count != 0
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:25]: the condition can be just: "emailTextField.text?.count != 0", or even better "!emailTextField.text?.isEmpty"

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    @IBOutlet weak var emailLabel: UILabel!
+    @IBOutlet weak var passwordLabel: UILabel!
+    @IBOutlet weak var notRegisteredLabel: UILabel!
+    @IBOutlet weak var validationLabel: UILabel!
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    @IBOutlet weak var loginButton: UIButton!
+    @IBAction func loginButton(_ sender: UIButton) {
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:22]: leave space between arguments and methods, also put buton actions below view did load

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        // validation for empty email textfield
+        guard let email = emailTextField.text, emailTextField.text?.count != 0
+        else{
+            validationLabel.isHidden = false
+            validationLabel.text = "Please enter Email"
+            return
+        }
+        
+        // validation for valid email format
+        if isValidEmail(emailAddress: email) == false {
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:33]: use !isValidEmail(emailAddress: email)

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            validationLabel.isHidden = false
+            validationLabel.text = "Email is invalid"}
+            else{
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:36]: always align else with if

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                validationLabel.isHidden = true
+            }
+        
+        //validation for empty password textfield
+        guard let password = passwordTextField.text,
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:41]: use "! passwordTextField.text?.isEmpty"

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        
+        // alert label is initially hidden from the user
+        validationLabel.isHidden = true
+        
+        //
+        emailTextField.delegate = self
+        passwordTextField.delegate = self
+        
+        // customization of the emailTextField to only have a black bottom border
+        let bottomLine1 = CALayer()
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:61]: textfield styling is duplicated, use a method and add the textfield as a parameter

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        // customization of the passwordTextField to only have a bottom black border
+        let bottomLine2 = CALayer()
+        bottomLine2.frame = CGRect(x: 0.0, y: passwordTextField.frame.height - 3, width: passwordTextField.frame.width, height: 1.0)
+        bottomLine2.backgroundColor = UIColor.black.cgColor
+        passwordTextField.borderStyle = UITextField.BorderStyle.none
+        passwordTextField.layer.addSublayer(bottomLine2)
+        passwordTextField.borderStyle = UITextField.BorderStyle.none
+        
+        // shadow of the login button
+        loginButton.layer.shadowColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.25).cgColor
```
Comment (iOSinternshipen) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:77]: this can be a method for button styling

**Review by iOSinternshipen — CHANGES_REQUESTED** [review]

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    @IBOutlet weak var loginLabel: UILabel!
+    @IBOutlet weak var emailLabel: UILabel!
+    @IBOutlet weak var passwordLabel: UILabel!
+    @IBOutlet weak var notRegisteredLabel: UILabel!
+    @IBOutlet weak var validationLabel: UILabel!
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    @IBOutlet weak var loginButton: UIButton!
```
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:21]: these should be marked as private (outlets/actions)

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift (hunk 1 above)
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:26]: leave a space between else and {

File: Internship2022TeamSzabi/Internship2022TeamSzabi/AppDelegate.swift [hunk 10]
```diff
@@ -14,6 +14,15 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
+        // ============================================================
+
+                self.window = UIWindow(frame: UIScreen.main.bounds)
+
+                self.window?.rootViewController = LoginViewController()
+               
+                self.window?.makeKeyAndVisible()
+
```
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/AppDelegate.swift:24]: no need to use self; indent accordingly

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift (hunk 5 above)
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:36]: space between else and {

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+            validationLabel.isHidden = false
+            validationLabel.text = "Email is invalid"}
+            else{
+                validationLabel.isHidden = true
+            }
+        
+        //validation for empty password textfield
+        guard let password = passwordTextField.text,
+                password.count != 0
+        else{
```
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:43]: spacing

**Review by IuliaRAnisoi — CHANGES_REQUESTED** [review]
In your storyboard, you should also remove the width constraints for the "email" and "password" labels. They already have leading and trailing constraints, and adding the width ones cause conflicting constraints errors.

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 12]
```diff
@@ -9,84 +9,95 @@ import UIKit
...
+        // validation for empty password textfield
+        guard !passwordTextField.text!.isEmpty
+        else {
             validationLabel.isHidden = false
-            validationLabel.text = "Please enter password"
+            validationLabel.text = "Please enter Password"
             return
-        }
+             }
     }
```
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:71]: please indent correctly by selecting this entire block and then use _control+i_

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 13]
```diff
@@ -9,84 +9,95 @@ import UIKit
...
+    @IBOutlet private var notRegisteredLabel: UILabel!
+    
+    @IBOutlet private var validationLabel: UILabel!
+    
+    @IBOutlet private var emailTextField: UITextField!
+    
+    @IBOutlet private var passwordTextField: UITextField!
+    
+    @IBOutlet private var loginButton: UIButton!
+
```
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:27]: you should remove the extra lines of spacing between variable declarations

File: Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift [hunk 14]
```diff
@@ -9,84 +9,95 @@ import UIKit
...
-            }
+             }
         
-        //validation for empty password textfield
-        guard let password = passwordTextField.text,
-                password.count != 0
-        else{
+        // validation for empty password textfield
+        guard !passwordTextField.text!.isEmpty
+        else {
```
Comment (IuliaRAnisoi) [inline Internship2022TeamSzabi/Internship2022TeamSzabi/Views/Login/LoginViewController.swift:66]: guard and else statements should be on the same line

**Review by IuliaRAnisoi — CHANGES_REQUESTED** [review]

**Review by IuliaRAnisoi — APPROVED** [review]

**Pull request merged**